views/pages/ViewChartPage/index.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from 'react'
+import qs from 'qs'
+import dayjs from 'dayjs'
+import timezone from 'dayjs/plugin/timezone'
+import advanced from 'dayjs/plugin/advancedFormat'
+import utc from 'dayjs/plugin/utc'
+import FileSaver from 'file-saver'
+import { useParams, useLocation, useNavigate } from 'react-router-dom'
+import { Box, Typography } from '@mui/material'
+
+import BarGraph from '../../components/BarGraph'
+import GraphTable from '../../components/GraphTable'
+import ChartFilterForm from '../../forms/ChartFilterForm'
+import PageHeader from '../../components/PageHeader'
+import ChartDescription from '../../components/ChartDescription'
+import { apiRoutes, routes } from '../../routes/routes'
+import api from '../../api'
+import { SITES_KEY } from '../../../server/constants'
+import { fontSize } from '../../../constants'
+
+import './ViewChartPage.css'
+
+dayjs.extend(utc)
+dayjs.extend(timezone)
+dayjs.extend(advanced)
+
+const ViewChartPage = () => {
+  const { search } = useLocation()
+  const { chart_id } = useParams()
+  const [graph, setGraph] = useState(null)
+  const [expanded, setExpanded] = useState(false)
+  const shortDescription = graph?.description?.substring(0, 296) + '...' || ''
+  const handleDescriptionExpand = () => setExpanded(!expanded)
+  const navigate = useNavigate()
+  const onSubmit = async (formValues) => {
+    const filters = Object.keys(formValues).reduce((filterObj, key) => {
```

Comment:
Can we make the argument to reduce a helper like `reduceFormToFilters` or something? I'm not picky about the name it's just easier to read IMO.

Reply:
[991acfb](https://github.com/AMP-SCZ/dpdash/pull/643/commits/991acfb8e98f59432e4140efeedb8f1bd35d280b)